### PR TITLE
DEVPROD-8861 Remove ssh cloning fallback

### PR DESCRIPTION
**Pull request by malikchaya2**
DEVPROD-8861

### Description
Don't fall back to ssh cloning because soon it won't be available as a fallback and everything should have a token.

### CODE REVIEW

File: operations/fetch.go [hunk 1]
```diff
@@ -205,6 +205,9 @@ type cloneOptions struct {
 
 func clone(opts cloneOptions) error {
 	// clone the repo first
+	if opts.token == "" {
```
Comment (Kimchelly) [inline operations/fetch.go:208]: If `evergreen fetch` requires a token to be passed in now, doesn't that mean the `evergreen fetch` that Evergreen runs on their behalf also has to pass some kind of token? Did we test that spawning a host with task data still works as expected?

File: agent/command/git.go [hunk 2]
```diff
@@ -219,8 +219,14 @@ func (opts cloneOpts) buildHTTPCloneCommand(forApp bool) ([]string, error) {
 	}
 	var gitURL string
 	if forApp {
+		if opts.token == "" {
```
Comment (Kimchelly) [inline agent/command/git.go:222]: Isn't the token already checked during [options validation](https://github.com/evergreen-ci/evergreen/blob/839e7fb0154aa9a9280615594b51440ad1b70bc0/agent/command/git.go#L123-L125)?

**Review by Kimchelly — CHANGES_REQUESTED** [review]

File: operations/fetch.go (hunk 1 above)
Comment (Kimchelly) [inline operations/fetch.go:208]: It also seems like this is something that could be validated earlier on in the CLI as a before func, so if they run `evergreen fetch --source`, they also have to set a `--token`.

Comment (malikchaya2) [reply operations/fetch.go:208]: I'm working on that for DEVPROD-8873. I thought I'd have the PR up already. Let me close this and re-open once ready.

**Pull request closed without merging**